### PR TITLE
Adding the TIM0 interrupt callback function 

### PR DESCRIPTION
**Pull request by diymat**
Adds the the TIM0 interrupt callback function.

### CODE REVIEW

File: hardware/arduino/avr/cores/arduino/wiring.c [hunk 1]
```diff
@@ -38,6 +38,9 @@
 volatile unsigned long timer0_overflow_count = 0;
 volatile unsigned long timer0_millis = 0;
 static unsigned char timer0_fract = 0;
+static (*TIM0_Callback)(void) = NULL;
+
+#define CALLBACKDIVIDER		16
```
Comment (per1234) [inline hardware/arduino/avr/cores/arduino/wiring.c]: The style convention established in this file and through much of the rest of the core is for the macro name to be separated from the definition by a single space.

**Comment by PaulStoffregen** [comment]
Interrupt-based callbacks are a pretty substantial Arduino API change.  Should probably be discussed....

**Comment by diymat** [comment]
> Interrupt-based callbacks are a pretty substantial Arduino API change.

IMO AVR-s have a very limited number of timers and using another one for something easy is a significant waste of the resources

**Comment by facchinm** [comment]
I'm closing this PR since it's very unlikely we'll ever support a non generic timer library. In that case, the proposal must be filed in https://github.com/arduino/ArduinoCore-api .

**Pull request closed without merging**